app/src/main/java/com/github/se/cyrcle/model/network/ConnectivityObserver.kt

```diff
@@ -0,0 +1,80 @@
+package com.github.se.cyrcle.model.network
+
+import android.content.Context
+import android.util.Log
+import com.github.se.cyrcle.model.map.MapViewModel
+import com.github.se.cyrcle.model.parking.ParkingViewModel
+import com.github.se.cyrcle.model.user.UserViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class ConnectivityObserver(
```

Comment:
Generally, for classes like this, a bit of documentation explaining what it is is appreciated! Non-blocking though